### PR TITLE
Fixes #18275: internally use a more secured version type for techniques version

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18275

### CODE REVIEW

**Comment by gpoblon** [comment]
Commit modified

**Comment by gpoblon** [comment]
Commit modified

File: rudder-lang/tests/techniques/simplest/technique.rl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 @format = 0
 @name = "simplest"
 @description = "rudderlang simplest for a complete loop"
-@version = 1
+@version = 1.0
```
Comment (peckpeck) [inline rudder-lang/tests/techniques/simplest/technique.rl:5]: shouldn't this be a string ie surrounded with " ?

**Comment by gpoblon** [comment]
PR updated with a new commit

**Review by peckpeck — APPROVED** [review]

File: rudder-lang/src/output.rs [hunk 2]
```diff
@@ -290,9 +290,9 @@ impl Logs {
                     .map_err(|e| format!("Building JSON output led to an error: {}", e))
                     .unwrap(); // dev error if this does not work
                 println!("{}", fmtoutput);
-                if is_success {
+                if !is_success {
                     eprintln!(
-                        "An error occurred, could not create content from '{}' because: '{}'",
+                        "An error occurred, could not create content frome '{}' because: '{}'",
```
Comment (amousset) [inline rudder-lang/src/output.rs:295]:
```suggestion
                        "An error occurred, could not create content from '{}' because: '{}'",
```

Comment (gpoblon) [reply rudder-lang/src/output.rs:295]: oh, don't know how it ended up here

**Comment by gpoblon** [comment]
PR updated with a new commit

**Comment by gpoblon** [comment]
OK, squash merging this PR

**Pull request merged**